libp2p/network/connection/swarm_connection.py

```diff
@@ -45,43 +51,32 @@ def __init__(self, muxed_conn: IMuxedConn, swarm: "Swarm") -> None:
             await stream.reset()
         # Force context switch for stream handlers to process the stream reset event we just emit
         # before we cancel the stream handler tasks.
-        await asyncio.sleep(0.1)
+        await trio.sleep(0.1)
 
-        for task in self._tasks:
-            task.cancel()
-            try:
-                await task
-            except asyncio.CancelledError:
-                pass
-        # Schedule `self._notify_disconnected` to make it execute after `close` is finished.
         self._notify_disconnected()
 
     async def _handle_new_streams(self) -> None:
+        self.event_started.set()
         while True:
             try:
                 stream = await self.muxed_conn.accept_stream()
             except MuxedConnUnavailable:
-                # If there is anything wrong in the MuxedConn,
-                # we should break the loop and close the connection.
                 break
             # Asynchronously handle the accepted stream, to avoid blocking the next stream.
-            await self.run_task(self._handle_muxed_stream(stream))
+            self.swarm.manager.run_task(self._handle_muxed_stream, stream)
```

Comment:
If you update to the latest `async-service` release (which you should), this *should* cause `mypy` to complain.  The `run_task` APIs are intentionally not exposed on the public API of `ManagerAPI`.

My recommendation would be **not** to use the `ManagerAPI` but instead to just open a local nursery here in this method and just use the raw nursery API to run this in the backgroung.